core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingSimulator.kt

```diff
@@ -13,51 +14,97 @@ import fr.sncf.osrd.utils.units.Offset
 interface ETCSBrakingSimulator {
     val context: EnvelopeSimContext
 
-    /** Compute the ETCS braking envelope for each LOA. */
+    /**
+     * Compute the ETCS braking envelope for each LoA and return a new envelope taking said curves
+     * into account.
+     */
     fun addSlowdownBrakingCurves(
         envelope: Envelope,
         limitsOfAuthority: Collection<LimitOfAuthority>
     ): Envelope
 
-    /** Compute the ETCS braking envelope for each EOA. */
+    /**
+     * Compute the ETCS braking envelope for each EoA and return a new envelope taking said curves
+     * into account.
+     */
     fun addStopBrakingCurves(
         envelope: Envelope,
         endsOfAuthority: Collection<EndOfAuthority>
     ): Envelope
+
+    /** Compute the ETCS braking curves for each LoA, ordered by LoA offset. */
+    fun computeSlowdownBrakingCurves(
+        mrsp: Envelope,
+        limitsOfAuthority: Collection<LimitOfAuthority>
+    ): LOAETCSBrakingCurves
+
+    /** Compute the ETCS braking curves for each EoA, ordered by EoA offset. */
+    fun computeStopBrakingCurves(
+        mrsp: Envelope,
+        endsOfAuthority: Collection<EndOfAuthority>
+    ): EOAETCSBrakingCurves
 }
 
+typealias ETCSBrakingCurves = EnumMap<BrakingCurveType, ETCSBrakingCurve?>
+
+typealias LOAETCSBrakingCurves = NavigableMap<LimitOfAuthority, ETCSBrakingCurves>
+
+typealias EOAETCSBrakingCurves = NavigableMap<EndOfAuthority, ETCSBrakingCurves>
+
+data class ETCSBrakingCurve(val brakingCurveType: BrakingCurveType, val brakingCurve: Envelope)
+
 data class LimitOfAuthority(
     val offset: Offset<TravelledPath>,
     val speed: Double,
-) {
+) : Comparable<LimitOfAuthority> {
     init {
         assert(speed > 0)
     }
+
+    override fun compareTo(other: LimitOfAuthority): Int {
+        return offset.compareTo(other.offset)
```

Comment:
Can we fully compare? It may complicate handling of wrong duplicate bugs (real duplicates would be weird but OK).

Ditto for EoA.
```suggestion
        if (offset != other.offset) {
            return offset.compareTo(other.offset)
        }
        return speed.compareTo(other.speed)
```

Reply:
I hate this. There shouldn't be LoAs at the same spot, ditto for EoAs, that's one of our core hypothesis for the computation of ETCS braking curves, if we don't have that pretty much all our calculations are bugged.

What about throwing an exception? If offsets are equal, throw an exception, that shouldn't be possible: we shouldn't try sorting a list of LoAs. It's not pretty either, but at least that way it gets the job done.

Reply:
Agreed on the exception, but it's not the role of the `compareTo`, is it?
And merging them through this compare would probably hide the bug, I guess.

Anyway, it's a best-effort thing :
* stronger opinion on not raising in compareTo.
* lighter opinion to have an "external" exception, and a complete compare.

Reply:
Ok let's just add a better comparison, and if at some point we hve a bug (improbable but we never know), we'll add in asserts.